runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -302,10 +301,18 @@ public void onExperienceChanged(ExperienceChanged event)
 		}
 
 		final XpUpdateResult updateResult = xpState.updateSkill(skill, currentXp, startGoalXp, endGoalXp);
-		final boolean updated = XpUpdateResult.UPDATED.equals(updateResult);
-		xpPanel.updateSkillExperience(updated, xpPauseState.isPaused(skill), skill, xpState.getSkillSnapshot(skill));
-		xpState.recalculateTotal();
-		xpPanel.updateTotal(xpState.getTotalSnapshot());
+		xpPanel.updateSkillExperience(updateResult == XpUpdateResult.UPDATED, xpPauseState.isPaused(skill), skill, xpState.getSkillSnapshot(skill));
+
+		if (skill == Skill.CONSTRUCTION && updateResult == XpUpdateResult.INITIALIZED)
```

Comment:
Hmm, if you reset construction wouldnt this cause reset of overall too? I would maybe flip this, e.g check for isInitialized(OVERALL) before checking for skill being construction

Reply:
Actually, if you reset skill this would never return initialized again until resetState method call, so I guess this is fine